apps/web/src/pages/templates/components/TemplateEditorFormProvider.tsx

```diff
@@ -100,7 +100,7 @@ const makeTemplateCopy = ({
 });
 
 export const makeVariantFromStep = (stepToVariant: IFormStep): IFormStep => {
-  const { id: _, variants, template, ...rest } = stepToVariant;
+  const { id: _, variants, template, _templateId, ...rest } = stepToVariant;
```

Comment:
previously, it was saving the _templateId of the root step, which lead to a scenario where the content of root step and newly created variant step was using the same message template